lib/cluster/ClusterOptions.ts

```diff
@@ -108,5 +128,6 @@ export const DEFAULT_CLUSTER_OPTIONS: IClusterOptions = {
   retryDelayOnClusterDown: 100,
   retryDelayOnTryAgain: 100,
   slotsRefreshTimeout: 1000,
-  slotsRefreshInterval: 5000
+  slotsRefreshInterval: 5000,
+  dnsLookup: lookup
```

Comment:
maybe wrap it as a promise here via `util.promisify` (I dont remember when it became available) or just as a

```ts
dnslookup(hostname: string): Promise<string> => {
  return new Promise((resolve, reject) => {
    lookup(hostname, (err, address) => {
          if (err) {
            debug('failed to resolve hostname %s to IP: %s', hostname, err.message)
            reject(err)
          } else {
            debug('resolved hostname %s to IP %s', hostname, address)
            resolve(address)
          }
        })
  });
},
```

Reply:
and then the code would be cleaner in the actual resolve function